tgui/packages/tgui/interfaces/Plantmaster.js

```diff
@@ -0,0 +1,414 @@
+ * @author Amylizzle (https://github.com/amylizzle)
+ * @license MIT
+ */
+
+import { useBackend, useSharedState, useLocalState } from "../backend";
+import { Button, Dimmer, Divider, Flex, NumberInput, Section, Box, Dropdown, Tabs, SectionEx, Stack, Table } from '../components';
+import { Window } from '../layouts';
+import { Fragment } from 'inferno';
+import { NoContainer, ReagentGraph, ReagentList } from './common/ReagentInfo';
+import { TableCell, TableRow } from "../components/Table";
+import { clamp } from 'common/math';
+import { capitalize } from './common/stringUtils';
+import { truncate } from '../format';
+
+export const Plantmaster = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    extractables,
+    seeds,
+    category,
+    category_lengths,
+    inserted,
+    inserted_container,
+    seedoutput,
+    splice_chance,
+    show_splicing,
+    splice_seeds,
+    sortBy,
+    sortAsc,
+  } = data;
+  return (
+    <Window
+      resizable
+      title="Plantmaster Mk4"
+      width={1100}
+      height={450}>
+      <Window.Content>
+        <Tabs>
+          <Tabs.Tab
+            selected={category === 'overview'}
+            onClick={() => {
+              act('change_tab', { 'tab': 'overview' });
+            }}>
+            Overview
+          </Tabs.Tab>
+          <Tabs.Tab
+            selected={category === 'extractables'}
+            onClick={() => {
+              act('change_tab', { 'tab': 'extractables' });
+            }}>
+            Seed Extraction {`(${category_lengths[0]})`}
+          </Tabs.Tab>
+          <Tabs.Tab
+            selected={category === 'seedlist'}
+            onClick={() => {
+              act('change_tab', { 'tab': 'seedlist' });
+            }}>
+            Seeds {`(${category_lengths[1]})`}
+          </Tabs.Tab>
+          <Tabs.Tab
+            backgroundColor={inserted_container !== null ? "green" : "blue"}
+            selected={inserted_container !== null}
+            icon="eject"
+            onClick={() => inserted_container !== null ? act('ejectbeaker') : act('insertbeaker')}
+            bold>
+            {inserted_container !== null ? inserted : "Insert Beaker"}
+          </Tabs.Tab>
+        </Tabs>
+
+
+        {category === 'overview' && <PlantOverview cat_lens={category_lengths} container={inserted ? inserted_container : null} seedoutput={seedoutput} />}
+        {category === 'extractables' && <PlantExtractables seedoutput={seedoutput} produce={extractables} sortBy={sortBy} sortAsc={sortAsc} />}
+        {category === 'seedlist' && <PlantSeeds seeds={seeds} seedoutput={seedoutput} splicing={show_splicing} splice_chance={splice_chance} splice_seeds={splice_seeds} sortBy={sortBy} sortAsc={sortAsc} />}
+      </Window.Content>
+    </Window>
+  );
+};
+
+const compare = function (a, b, sortBy, sortAsc) {
+  if (sortBy === "name" || sortBy === "species")
+  {
+    if (sortAsc) {
+      return (''+a[sortBy]).localeCompare(b[sortBy]);
```

Comment:
Just a little cleaner instead of using the JS quirks with types. Especially if someone ever wants to move this to typescript at some point

```suggestion
      return (a[sortBy] ?? '').localeCompare(b[sortBy] ?? '');
```